public/app/features/browse-dashboards/state/hooks.ts

```diff
@@ -83,3 +95,38 @@ function createFlatTree(
 
   return items.flatMap((item) => mapItem(item, folderUID, level));
 }
+
+/**
+ * Returns a DashboardTreeSelection but removes any selected folder's children.
+ * This is useful when making backend requests to move or delete items.
+ * In this case, we only need to move/delete the parent folder and it will cascade to the children.
+ * @param selectedItems Overall selection state
+ * @param childrenByParentUID Arrays of children keyed by their parent UID
+ */
+function getSelectedItemsForBackend(
+  selectedItems: DashboardTreeSelection,
+  childrenByParentUID: Record<string, DashboardViewItem[] | undefined>
+): Omit<DashboardTreeSelection, 'panel'> {
+  // Take a copy of the selected items to work with
+  // We don't care about panels here, only dashboards and folders can be moved or deleted
+  const result: Omit<DashboardTreeSelection, 'panel'> = {
+    dashboard: { ...selectedItems.dashboard },
+    folder: { ...selectedItems.folder },
+  };
+
+  for (const [parentFolderUID, children] of Object.entries(childrenByParentUID)) {
+    // Check if the parent folder is selected
+    if (selectedItems.folder[parentFolderUID]) {
+      // Unselect any children
+      children?.forEach((child) => {
+        if (child.kind === 'dashboard') {
+          result.dashboard[child.uid] = false;
+        }
+        if (child.kind === 'folder') {
+          result.folder[child.uid] = false;
+        }
+      });
```

Comment:
```suggestion
    // Unselect any children
    for (const child of children ?? []) {
      if (child.kind === 'dashboard') {
        result.dashboard[child.uid] = false;
      }
      if (child.kind === 'folder') {
        result.folder[child.uid] = false;
      }
    }
```

Annoying nit: I'm trying to prefer for-of/for-in over Array.forEach, especially when the forEach callback isn't a 'pure' function (it modifies `result` outside it's scope.